ai/Algorithm/path_partitionner.py

```diff
@@ -165,14 +181,33 @@ def update_last_path(self, start, target):
             self.old_path.points[-1] = target
             return self.old_path
         else:
-            return self.path_planner(start.array, target.array)
+            if self.points_to_pass_by is None:
+                return self.path_planner(start.array, target.array)
+            else:
+                return self.generate_path_through_way_points(start, target)
 
     def verify_first_points_of_path(self, path):
         if len(path.points) > 2:
             if not self.is_path_colliding(path.start.array, path.points[2].array):
                 del path.points[1]
         return path
 
+    def generate_path_through_way_points(self, start, target):
+        start = start
+        point = self.points_to_pass_by[0]
+        sub_paths = PathPartitionner().get_path(start, point, self.obstacles, ball_collision=self.ball_collision[0])
+        start = point
+        if len(self.points_to_pass_by) > 1:
+            for point, ball_collision in zip(self.points_to_pass_by[1:], self.ball_collision[1:]):
```

Comment:
Nitpick: Évite d'avoir une variable start à géré
`for point, ball_collision, prev_point in zip(self.points_to_pass_by[1:], self.ball_collision[1:], self.points_to_pass_by):`